packages/monaco-editor-wrapper/src/wrapper.ts

```diff
@@ -33,11 +33,27 @@ export type EditorConfig = {
     diffEditorOptions?: editor.IStandaloneDiffEditorConstructionOptions;
 }
 
+export type WebSocketStartOptions = {
+    /** Adds handle on starting of languageClient */
+    onStart: () => void;
+    /** Reports Status Of Language Client */
+    reportStatus?: boolean;
+}
+
+export type WebSocketStopOptions = {
+    /** Adds handle on stopping of languageClient */
+    onStop: () => void;
+    /** Reports Status Of Language Client */
+    reportStatus?: boolean;
+}
+
 export type LanguageClientConfig = {
     enabled: boolean;
     useWebSocket?: boolean;
     webSocketConfigOptions?: WebSocketConfigOptions;
     workerConfigOptions?: WorkerConfigOptions;
+    webSocketStartOptions?: WebSocketStartOptions;
```

Comment:
I think it makes sense to move `webSocketStartOptions` and `webSocketStopOptions` into `webSocketConfigOptions`.